net-utils/src/sockets.rs

```diff
@@ -55,3 +57,35 @@ pub fn bind_gossip_port_in_range(
         bind_common_in_range_with_config(bind_ip_addr, port_range, config).expect("Failed to bind")
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use {super::*, std::env};
+
+    #[test]
+    // This overlay function is needed for tests that relay on NEXTEST_TEST_GLOBAL_SLOT environment variable.
+    fn test_localhost_port_range() {
+        test_localhost_port_range_without_nextest();
+        test_localhost_port_range_with_nextest();
+    }
+
+    fn test_localhost_port_range_without_nextest() {
```

Comment:
Calling this test just once does not check that we are actually allocating non-overlapping ranges when this is called several times.